components/script/dom/document.rs

```diff
@@ -1219,6 +1224,80 @@ impl Document {
         }
     }
 
+    pub(crate) fn show_select_element_menu_for(
```

Comment:
`document.rs` is getting really large, so maybe you can move this to `htmlselectelement.rs` and just make sure that `fn send_to_embedder` is `pub(crate)`?